src/bio/bio_buffer.c

```diff
@@ -865,9 +865,11 @@ dma_drop_iod(struct bio_dma_buffer *bdb)
 	D_ASSERT(bdb->bdb_active_iods > 0);
 	bdb->bdb_active_iods--;
 
-	ABT_mutex_lock(bdb->bdb_mutex);
-	ABT_cond_broadcast(bdb->bdb_wait_iods);
-	ABT_mutex_unlock(bdb->bdb_mutex);
+	if (bdb->bdb_waiters) {
+		ABT_mutex_lock(bdb->bdb_mutex);
```

Comment:
This 'dbd_mutex' can be removed. The DMA buffer is per-xstream, and SPDK io completion always happen on the same xstream where the io is issued, so the mutex isn't necessary.